activerecord/lib/active_record/autosave_association.rb

```diff
@@ -302,7 +302,7 @@ def nested_records_changed_for_autosave?
       def validate_single_association(reflection)
```

Comment:
```suggestion
      def validate_single_association(reflection)
           record = association_instance_get(reflection.name)&.reader
           association_valid?(reflection, record) if record&.changed_for_autosave? || custom_validation_context?
        end
```

Reply:
@gklsan I'm not sure I follow your suggestion here